pkg/segment/results/mresults/seriesresult.go

```diff
@@ -404,37 +391,42 @@ func ApplyRangeFunction(ts map[uint32]float64, function structs.Function) (map[u
 		// Rate at edge does not exist.
 		delete(ts, sortedTimeSeries[0].downsampledTime)
 		return ts, nil
-	case segutils.IRate:
-		// Calculate the instant rate (per-second rate) for each timestamp, based on the last two data points within the timewindow
-		// If the previous point is outside the time window, we still need to use it to calculate the current point's rate, unless its value is greater than the value of the current point
-		if len(sortedTimeSeries) == 0 {
-			return nil, nil
+	case segutils.Increase:
+		// Increase is extrapolated to cover the full time range as specified in the range vector selector. (increse = avg rate * timewindow)
+		ts := evaluateRate(sortedTimeSeries, ts, timeWindow)
+		for key, rateVal := range ts {
+			ts[key] = rateVal * float64(timeWindow)
 		}
+		return ts, nil
+	case segutils.Delta:
+		// Calculates the difference between the first and last value of each time series element within the timewindow
+		for i := 1; i < len(sortedTimeSeries); i++ {
+			timeWindowStartTime := sortedTimeSeries[i].downsampledTime - timeWindow
+			preIndex := sort.Search(len(sortedTimeSeries), func(j int) bool {
+				return sortedTimeSeries[j].downsampledTime >= timeWindowStartTime
+			})
 
-		var dx, dt float64
+			if i <= preIndex { // Can not find the second point within the time window
+				delete(ts, sortedTimeSeries[i].downsampledTime)
+				continue
+			}
+			ts[sortedTimeSeries[i].downsampledTime] = sortedTimeSeries[i].dpVal - sortedTimeSeries[preIndex].dpVal
```

Comment:
Do you have to handle resets with `Delta`?

Reply:
- Rate should only be used with counters and native histograms where the components behave like counters
- Delta should only be used with gauges and native histograms where the components behave like gauges (so-called gauge histograms).

Based on [doc](https://prometheus.io/docs/concepts/metric_types/)
> A counter is a cumulative metric that represents a single [monotonically increasing counter], while gauge can arbitrarily go up and down.

So i think we do not need to handle reset logic for a gauge metric